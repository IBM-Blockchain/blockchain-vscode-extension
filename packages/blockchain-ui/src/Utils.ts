import IVSCode from './interfaces/IVSCode';
declare const vscode: IVSCode;

const Utils: any = {

    changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    },

    postToVSCode(message: {command: string, data?: any}): void {
        vscode.postMessage(message);
    }

};

export default Utils;
