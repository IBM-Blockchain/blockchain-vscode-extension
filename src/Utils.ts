const Utils: any = {

    changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    }

};

export default Utils;
