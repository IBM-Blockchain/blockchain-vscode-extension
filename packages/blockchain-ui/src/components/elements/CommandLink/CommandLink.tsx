import React, { Component } from 'react';
import { Link } from 'carbon-components-react';
import Utils from '../../../Utils';

interface IProps {
    commandName: string;
    linkContents: string;
    className?: string;
    commandData?: any[];
    id?: string;
}

class CommandLink extends Component<IProps> {
    constructor(props: Readonly<IProps>) {
        super(props);
        this.runCommand = this.runCommand.bind(this);
    }

    runCommand(): void {
        const message: {command: string, data?: any} = {
            command: this.props.commandName
        };
        if (this.props.commandData) {
            message.data = this.props.commandData;
        }

        Utils.postToVSCode(message);
    }

    render(): JSX.Element {
        const style: string = this.props.className ? this.props.className : '';

        return (
            <Link className={style} href='' onClick={this.runCommand} id={this.props.id}>
                {this.props.linkContents}
            </Link>
        );
    }

}

export default CommandLink;
