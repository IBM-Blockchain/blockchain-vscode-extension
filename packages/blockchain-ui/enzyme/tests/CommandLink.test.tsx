import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import CommandLink from '../../src/components/elements/CommandLink/CommandLink';
import Utils from '../../src/Utils';

chai.should();
chai.use(sinonChai);

describe('CommandLink component', () => {
    let mySandBox: sinon.SinonSandbox;
    let postToVSCodeStub: sinon.SinonStub;

    beforeEach(() => {
        mySandBox = sinon.createSandbox();
        postToVSCodeStub = mySandBox.stub(Utils, 'postToVSCode').resolves();
    });

    afterEach(() => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<CommandLink linkContents='my link' commandName='my command'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should call a command in VS Code when clicked', () => {
        const component: any = mount(<CommandLink linkContents='my link' commandName='my command'/>);
        component.find('a').simulate('click');
        postToVSCodeStub.should.have.been.calledOnceWithExactly({
            command: 'my command'
        });
    });

    it('should post additional information to VS Code when provided', () => {
        const component: any = mount(<CommandLink linkContents='my link' commandName='my command' commandData={['additional', 'information']}/>);
        component.find('a').simulate('click');
        postToVSCodeStub.should.have.been.calledOnceWithExactly({
            command: 'my command',
            data: ['additional', 'information']
        });
    });

    it('should add any additional styles provided', () => {
        const component: any = mount(<CommandLink linkContents='my link' commandName='my command' className='some-style'/>);
        component.find('a').hasClass('some-style').should.equal(true);
    });
});
