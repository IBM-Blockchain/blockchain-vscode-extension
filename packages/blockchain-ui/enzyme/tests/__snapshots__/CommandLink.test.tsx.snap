// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandLink component should render the expected snapshot 1`] = `
<a
  className="bx--link"
  href=""
  onClick={[Function]}
>
  my link
</a>
`;
